src/lib/helpers/deployment.helper.ts

```diff
@@ -288,3 +302,25 @@ export function waitForBatchedPendingTransactions(
 
   return concat(transactions$, receipts$);
 }
+
+export function getContractAddressFromReceipt(
+  receipt: ethers.providers.TransactionReceipt,
+  isSignerUniversalProfile: boolean
+) {
+  return isSignerUniversalProfile
+    ? receipt.contractAddress ||
+        ethers.utils.defaultAbiCoder.decode(['address'], receipt.logs[1].topics[2]).toString()
+    : receipt.contractAddress || receipt.to;
+}
+
+export function getContractAddressFromReceipt$(
+  accountDeploymentReceipt$: Observable<DeploymentEventBase>,
+  isSignerUniversalProfile$: Observable<boolean>
+) {
+  return forkJoin([accountDeploymentReceipt$.pipe(takeLast(1)), isSignerUniversalProfile$]).pipe(
+    switchMap(async ([result, isSignerUniversalProfile]) => {
+      return getContractAddressFromReceipt(result.receipt, isSignerUniversalProfile);
+    }),
+    shareReplay()
```

Comment:
I m not really sure to understand how this function gets the contract address from receipt.
The name is very clear to wht the function does but I'm confused why the function calls itself. And also why we share replay the result of the function.

UPDATE
I think I understood why to use ShareReplay : Is it because we call this function several time in different cases and we want to access it results without having to call it again ?

Reply:
yes shareReplay means the function will not be called again and the result is available anywhere it is resolved